src/pages/CoSoulExplorePage/CoSoulItem.tsx

```diff
@@ -0,0 +1,83 @@
+import { AppLink, Avatar, Box, Flex, Image, Text } from '../../ui';
+
+import { CoSoul } from './useInfiniteCoSouls';
+
+const s3bucket = process.env.REACT_APP_S3_BASE_URL;
+
+export const CoSoulItem = ({ cosoul }: { cosoul: CoSoul }) => {
+  return (
+    <AppLink to={`/cosoul/${cosoul.address}`}>
+      <Box
+        key={cosoul.id}
+        css={{
+          overflow: 'hidden',
+          borderRadius: '$4',
+          position: 'relative',
+          '&:hover': {
+            cursor: 'pointer',
+            transition: 'all 0.2s ease-in-out',
+            transform: 'scale(1.05)',
+          },
+        }}
+      >
+        <Box
+          css={{
+            width: '100%',
+            aspectRatio: '1 / 1',
+          }}
+        >
+          <Image
+            css={{
+              width: '100%',
+              aspectRatio: '1 / 1' /*, top: 0, left: 0, position: 'absolute'*/,
+            }}
+            src={`${s3bucket}/cosoul/screenshots/${cosoul.token_id}.png`}
+            alt="CoSoul Screenshot"
+          />
+        </Box>
+        <Flex
+          css={{
+            p: '$sm',
+            width: '100%',
+            justifyContent: 'space-between',
+            alignItems: 'flex-start',
+          }}
+        >
+          <Flex
+            css={{
+              gap: '$sm',
+              alignItems: 'center',
+              flexGrow: 0,
+              flexShrink: 1,
+            }}
+          >
+            <Avatar
+              path={cosoul.profile_public?.avatar}
+              size="small"
+              name={cosoul.profile_public?.name ?? 'New'}
+            />
+            {/*TODO: if the name is long here this doesn't properly shrink */}
```

Comment:
Maybe use this and find the length that fits right?

```ts
function abbreviateString(str: string, maxLength: number): string {
  if (str.length <= maxLength) {
    return str;
  }
  return str.substring(0, maxLength - 3) + '...';
}
```

Reply:
did it